orderbook/src/fee.rs

```diff
@@ -97,6 +100,19 @@ impl MinFeeCalculator {
         amount: Option<U256>,
         kind: Option<OrderKind>,
     ) -> Result<Measurement, MinFeeCalculationError> {
+        if self.unsupported_tokens.contains(&sell_token) {
+            return Err(MinFeeCalculationError::UnsupportedToken(sell_token));
+        }
+
+        if buy_token
+            .map(|t| self.unsupported_tokens.contains(&t))
+            .unwrap_or_default()
+        {
+            return Err(MinFeeCalculationError::UnsupportedToken(
+                buy_token.expect("Must exist"),
+            ));
+        }
```

Comment:
```suggestion
        if let Some(token) = buy_token {
            if self.unsupported_tokens.contains(&token) {
                return Err(MinFeeCalculationError::UnsupportedToken(token));
            }
        }
```

Reply:
Right, I felt the upper was easier to merge with #542